docs/extensions/redirects.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: MIT
+
+import json
+import os
+from typing import Dict
+
+from _types import SphinxExtensionMeta
+from sphinx.application import Sphinx
+
+
+def main(app: Sphinx, exception: Exception) -> None:
+    # mapping of html node id (i.e., thing after "#" in URLs) to the same but
+    # prefixed with the right doc name
+    # e.g, api.html#disnake.Thread => api/channels.html#disnake.Thread
+    actual_redirects: Dict[str, str] = {}
+
+    for domainname, domain in app.env.domains.items():
+        if domainname != "py":
+            continue  # skip non-python objects
+
+        # first is the real object name - we dont need it
+        # second is object's "display name" (idk what is that mmlol), we dont need it
+        # third is object type (class, method etc), we too dont need it
+        # sixth is whether the object is aliased (not sure what does it actually mean,
+        # i guess whether you did :ref:`abcedfg <hijklm>` instead of just :ref:`hijklm`
+        # idk), again, we dont need it
+        for _, _, _, document, html_node_id, _ in domain.get_objects():
+            actual_redirects[html_node_id] = document + ".html#" + html_node_id
+
+    with open(os.path.join(app.outdir, "_static", "api_redirect.js"), "a") as redirects_js:
```

Comment:
should first ensure the file exists, as a sanity check

Reply:
Done in [`d8070db`](https://github.com/DisnakeDev/disnake/pull/722/commits/d8070db1f45fdb46fc3264db52d117c2ad5e86f6)